addons/hr_holidays/models/hr_leave.py

```diff
@@ -561,6 +566,10 @@ def _get_duration(self, check_leave_type=True, resource_calendar=None):
         """
         self.ensure_one()
         resource_calendar = resource_calendar or self.resource_calendar_id
+        resource_calendar = resource_calendar.with_context({
+            **self.env.context,
+            'employee_timezone': timezone(resource_calendar.tz),
+        })
```

Comment:
should be equivalent to:

```suggestion
        resource_calendar = resource_calendar.with_context(
            employee_timezone=timezone(resource_calendar.tz)
        )
```
unless the `resource_calendar` could have a different context than `self` that we want to override.

also on the line below, it seems that resource_calendar can be falsy, then `resource_calendar.tz` would be `False` which cause an error:

```py
In [8]: self.env['hr.leave'].resource_calendar_id.tz
Out[8]: False

In [9]: timezone(self.env['hr.leave'].resource_calendar_id.tz)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File ~/.local/venv/16.0/lib/python3.11/site-packages/pytz/__init__.py:170, in timezone(zone)
    167 if zone is None:
    168     raise UnknownTimeZoneError(None)
--> 170 if zone.upper() == 'UTC':
    171     return utc
    173 try:

AttributeError: 'bool' object has no attribute 'upper'
```
so it should only be one ifthere is a resource_calendar it seems.